pkg/services/ngalert/writer/prom.go

```diff
@@ -66,9 +66,15 @@ func PointsFromFrames(name string, t time.Time, frames data.Frames, extraLabels
 
 	points := make([]Point, 0, len(col.Refs))
 	for _, ref := range col.Refs {
-		fp, empty, _ := ref.NullableFloat64Value()
-		if empty || fp == nil {
-			return nil, fmt.Errorf("unable to read float64 value")
+		fp, empty, err := ref.NullableFloat64Value()
+		if empty {
+			return nil, fmt.Errorf("empty frame")
```

Comment:
Technically these are all for the field within the frame.

Looking back at this, `empty` is a poor name for this variable. This is true if the values in the field has length of anything except 1.

Reply:
The CR reader comment:
> a bool that is true if the value is empty (no field, or zero length field)
does this mean that is also wrong?

Reply:
This is the check:
```go
if n.ValueField.Len() != 1 {
	return nil, true, nil
}
```

But without digging into it, I think we reject the frame before we reach this point if the value is length > 1